client/src/actors.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use riker::actors::*;
+
+use std::{fmt::Debug, path::PathBuf};
+
+use engine::vault::{BoxProvider, Key, RecordHint, RecordId};
+
+use crate::{
+    bucket::Bucket,
+    client::{ClientMsg, InternalResults},
+    ids::VaultId,
+    key_store::KeyStore,
+    line_error,
+    provider::Provider,
+    snapshot::Snapshot,
+};
+
+/// Messages used to talk to the Bucket Actor.
+#[derive(Debug, Clone)]
+pub enum BMsg<P: BoxProvider + Debug> {
+    CreateVault(VaultId, Key<P>),
+    ReadData(Key<P>, RecordId),
+    WriteData(Key<P>, RecordId, Vec<u8>, RecordHint),
+    InitRecord(Key<P>, VaultId),
+    RevokeData(Key<P>, RecordId),
+    GarbageCollect(Key<P>),
+    ListAsk(Key<P>),
+    WriteSnapshot(String, Option<String>, Option<PathBuf>),
```

Comment:
I understand what most of the enum values means, but in some cases maybe named arguments would be better? I have no idea what is the Option<String> here without looking at the rest of the code.

Reply:
Will put documentation above each of the variant types.  For instance, in this one, `WriteSnapshot` takes the password, an optional Filename, and the Optional File Path.